pkg/appgw/requestroutingrules_test.go

```diff
@@ -463,4 +463,85 @@ var _ = Describe("Test routing rules generations", func() {
 			Expect(len(*configBuilder.appGw.URLPathMaps)).To(Equal(0))
 		})
 	})
+
+	Context("test path-based rule with 1 ingress and multiple service having with same paths", func() {
+		configBuilder := newConfigBuilderFixture(nil)
+		testServiceName := "testService"
+		endpoint1 := tests.NewEndpointsFixture()
+		endpoint2 := tests.NewEndpointsFixture()
+		endpoint2.Name = testServiceName
+		service1 := tests.NewServiceFixture(*tests.NewServicePortsFixture()...)
+		service2 := tests.NewServiceFixture(*tests.NewServicePortsFixture()...)
+		service2.Name = testServiceName
+
+		// 2 path based rules with path - /api1, /api2
+		ingressPathBased := tests.NewIngressFixture()
+		ingressPathBased.Annotations[annotations.SslRedirectKey] = "false"
+		backendBasic := tests.NewIngressBackendFixture(service2.Name, 80)
+
+		// Adding duplicate path /api for a different service in same ingress
+		duplicatePathRule := tests.NewIngressRuleFixture(tests.Host, "/api1", *backendBasic)
```

Comment:
can you add additional cases to this test covering:
1. duplicate path in same rule under different path [not covered]
1. duplicate path in different rules [covered]
1. duplicate path in different ingresses [not covered]

Reply:
Done